configgtm-v1/domain.go

```diff
@@ -0,0 +1,336 @@
+        LoadImbalancePercentage         float64           `json:"loadImbalancePercentage,omitempty"`
+        DefaultHealthMax                int               `json:"defaultHealthMax,omitempty"`
+        LastModified                    string            `json:"lastModified,omitempty"`
+        Status                          *ResponseStatus   `json:"status,omitempty"`
+        MapUpdateInterval               int               `json:"mapUpdateInterval,omitempty"`
+        MaxProperties                   int               `json:"maxProperties,omitempty"`
+        MaxResources                    int               `json:"maxResources,omitempty"`
+        DefaultSslClientPrivateKey      string            `json:"defaultSslClientPrivateKey,omitempty"`
+        DefaultErrorPenalty             int               `json:"defaultErrorPenalty,omitempty"`
+        Links                           []*Link           `json:"links,omitempty"`
+        Properties                      []*Property       `json:"properties,omitempty"`
+        MaxTestTimeout                  float64           `json:"maxTestTimeout,omitempty"`
+        CnameCoalescingEnabled          bool              `json:"cnameCoalescingEnabled,omitempty"`
+        DefaultHealthMultiplier         int               `json:"defaultHealthMultiplier,omitempty"`
+        ServermonitorPool               string            `json:"servermonitorPool,omitempty"`
+        LoadFeedback                    bool              `json:"loadFeedback,omitempty"`
+        MinTTL                          int64             `json:"minTTL,omitempty"`
+        GeographicMaps                  []*GeoMap         `json:"geographicMaps,omitempty"`
+        CidrMaps                        []*CidrMap        `json:"cidrMaps,omitempty"`
+        DefaultMaxUnreachablePenalty    int               `json:"defaultMaxUnreachablePenalty,omitempty"`
+        DefaultHealthThreshold          int               `json:"defaultHealthThreshold,omitempty"`
+        LastModifiedBy                  string            `json:"lastModifiedBy,omitempty"`
+        ModificationComments            string            `json:"modificationComments,omitempty"`
+        MinTestInterval                 int               `json:"minTestInterval,omitempty"`
+        PingPacketSize                  int               `json:"pingPacketSize,omitempty"`
+        DefaultSslClientCertificate     string            `json:"defaultSslClientCertificate,omitempty"`
+}
+
+type DomainsList struct {
+        DomainItems              []*DomainItem     `json:"items"`
+}
+
+type DomainItem struct {
+        AcgId                    string            `json:"acgId"`
+        LastModified             string            `json:"lastModified"`
+        Links                    []*Link           `json:"links"`
+        Name                     string            `json:"name"`
+        Status                   string            `json:"status"`
+}
+
+func SetHeader(req *http.Request) {
```

Comment:
Should this be exported? Or private to the package (e.g. `setHeader()`). You might also consider calling it `setVersionHeader` and allow passing in an argument for the version if you anticipate that changing?

Reply:
corrected